blog/_posts/2022-08-08-managing-tunnels-with-argocd.md

```diff
@@ -0,0 +1,236 @@
+
+To access the control plane of each tunnel server we need to set up Kubernetes Ingress. Since we will be using the `inlets-pro` Helm chart to deploy our tunnel servers you can follow the steps in "Install the prerequisites" and "Install an Issuer" over in the guide: [Use your Kubernetes cluster for exit-servers](https://github.com/inlets/inlets-pro/tree/master/chart/inlets-pro).
+
+After you completed setting up Ingress for the cluster we will install Argo CD. If you followed the previous steps, `arkade` should be available on your system. You can use it to install Argo CD.
+```bash
+arkade install argocd
+```
+
+## Run a TCP tunnel server
+To install a tunnel server we need to override the default values from the inlets-pro Helm chart. Argo CD does not allow the use of external values.yaml files. The values files must be in the same repository as the Helm chart. We can work around this limitation by creating an "umberella chart".
+
+```
+# Example umbrella chart
+├── Chart.yaml
+└── values-postgresql-customer1.yaml
+```
+
+In the `Chart.yaml` of the umbrella chart we specify the `inlets-pro` chart as a dependency. We can than add multiple values files, one for each server that has to be created. The different values files in the chart can be referenced later by the Argo CD apps for each tunnel server.
+```yaml
+apiVersion: v2
+name: inlets
+description: Helm chart for an inlets-pro TCP server
+type: application
+version: 0.1.0
+appVersion: "0.1.0"
+dependencies:
+  - name: inlets-pro
+    version: 0.4.0
+    repository: https://inlets.github.io/inlets-pro/charts
+```
+
+The `values-postgresql-customer1.yaml` sets the configuration values for the first tunnel server:
+```yaml
+inlets-pro:
+  tokenSecretName: postgresql-customer1-token
+  ingress:
+    domain: postgresql-customer1.example.com
+
+  dataPlane:
+    type: ClusterIP
+    ports:
+    - targetPort: 5432
+      protocol: TCP
+      name: postgresql
+      port: 5432
+```
+
+We set the address, in this case `postgresql-customer1.example.com`, that we want our inlets clients to connect to in the `ingress.domain` field. When the chart is deployed a Certificate is created and cert-manager will request a TLS certificate from Let’s Encrypt using a HTTP01 challenge.
+
+In the `tokenSecretName` field we set the name of secret containing the token that we want to use for this server. In the next section we will see how to create these these secrets and make them available in the cluster.
+
+## Manage tokens for your tunnels
+Each inlets tunnel server requires a unique token. This token needs to be stored as a secret in the cluster so that it can be accessed by the inlets server. Argo CD is un-opinionated about how secrets are managed. Wether you want to use Vault, Helm Secrets or SOPS, to name a few, they can all be set up to work with ArgoCD. We will be using [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets).
```

Comment:
SOPS is an acronym, therefore write it in full followed by (SOPS) in brackets. And add a hyperlink to the project

Reply:
Same for Vault